#This dataset is about heart failure clinical records data

#It is a Labelled dataset with Categorical output, Hence Decision Tree Classification Algorithm will be implemented

#1. Start by importing all necessary modules

In [10]:
import pandas as pd
import numpy as np
from sklearn.metrics import accuracy_score
from sklearn import metrics

#2. Import dataset

In [12]:
data = pd.read_csv(r'D:\DataScience\MyProjects\heart_failure_clinical_records_dataset.csv')
df=pd.DataFrame(data)
print(df.to_string())

        age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  smoking  time  DEATH_EVENT
0    75.000        0                       582         0                 20                    1  265000.00              1.90           130    1        0     4            1
1    55.000        0                      7861         0                 38                    0  263358.03              1.10           136    1        0     6            1
2    65.000        0                       146         0                 20                    0  162000.00              1.30           129    1        1     7            1
3    50.000        1                       111         0                 20                    0  210000.00              1.90           137    1        0     7            1
4    65.000        1                       160         1                 20                    0  327000.00              2.70          

#3. Select features for x and y

In [13]:
x =df.iloc[:, :-1].values # independent variable
y =df.iloc[:, -1].values # dependant variable

#4. Splitting the dataset into the Training set and Test set

In [15]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test= train_test_split(x, y, test_size= 0.25, random_state=0)

#5. Feature Scaling

In [16]:
from sklearn.preprocessing import StandardScaler
st_x= StandardScaler()
x_train= st_x.fit_transform(x_train)
x_test= st_x.transform(x_test)

#6. Fitting Decision Tree classifier to the training set

In [17]:
from sklearn.tree import DecisionTreeClassifier
classifier= DecisionTreeClassifier(criterion='gini', random_state=0)
classifier.fit(x_train, y_train)

DecisionTreeClassifier(random_state=0)

#7.From sklearn.tree import DecisionTreeClassifier

In [18]:
classifier= DecisionTreeClassifier(criterion='gini', random_state=0)
classifier.fit(x_train, y_train)
y_pred= classifier.predict(x_test)
df2=pd.DataFrame({"Actual Y_Test":y_test,"Prediction Data":y_pred})
print("Prediction Result")
print(df2.to_string())

Prediction Result
    Actual Y_Test  Prediction Data
0               0                0
1               0                0
2               1                1
3               0                0
4               0                0
5               0                0
6               0                1
7               0                0
8               1                0
9               0                0
10              0                0
11              1                1
12              1                1
13              1                0
14              1                1
15              1                1
16              0                0
17              0                0
18              0                0
19              0                0
20              0                1
21              0                1
22              1                0
23              0                0
24              1                1
25              0                0
26              1                1
27

#8. Evaluating Algorithm

In [19]:
print('Mean Absolute Error:', metrics.mean_absolute_error(y_test,y_pred))
print('Mean Squared Error:', metrics.mean_squared_error(y_test,y_pred))
print('Root Mean Squared Error:', np.sqrt(metrics.mean_squared_error(y_test, y_pred)))

Mean Absolute Error: 0.22666666666666666
Mean Squared Error: 0.22666666666666666
Root Mean Squared Error: 0.4760952285695233


#9. Evaluating Predictions

In [20]:
accuracy = accuracy_score(y_test, y_pred)
print('Accuracy: %.2f' % (accuracy*100))

Accuracy: 77.33


# conclusion

#The Observed accuracy for this model is 77.33 Hence this is considered as a good model.